packages/astro/src/compiler/transform/head.ts

```diff
@@ -21,6 +21,16 @@ export default function (opts: TransformOptions): Transformer {
         },
         InlineComponent: {
           enter(node) {
+            if (hasComponents) {
+              return
+            }
+
+            if (node.attributes && node.attributes.some(({ name }: any) => name.startsWith('client:'))) {
+              hasComponents = true;
+              return;
+            }
+
+            /** Check for legacy hydration */
             const [_name, kind] = node.name.split(':');
             if (kind && !hasComponents) {
```

Comment:
this `!hasComponents` will now never be hit due to the early return, would you mind removing this?

Reply:
Ah yep I missed that one, thanks for catching that!

Reply:
Removed in 111744b70e2e2fd54422609df5896b46117e71d5